view/light.twig

```diff
@@ -37,43 +37,79 @@
+        <h2 class="sr-only">{% trans sidebarHeading %}</h2>
         <ul class="nav nav-tabs{% if vocab.config.showChangeList and vocab.config.groupClassURI and vocab.config.showAlphabeticalIndex %} nav-four-wide{% endif %}">
         {% if request.page != 'search' %}
           {% if request.vocab.config.showAlphabeticalIndex %}
+          <h3 class="sr-only">{% trans "Assort vocabulary concepts alphabetically" %}</h3>
           <li id="alpha"{% if search_results|length != 1 or (term and search_results|length == 1) %} class ="active"{% endif %}><a href="{{ request.vocabid }}/{{ request.lang }}/index{% if request.contentLang != request.lang %}?clang={{ request.contentLang }}{% endif %}">{% if vocab.config.showChangeList and vocab.config.groupClassURI %}{% trans'A-Z'%}{% else %}{% trans "Alpha-nav" %}{% endif %}</a></li>
           {% endif %}
-          <li id="hierarchy{% if not vocab.config.showHierarchy and not search_results|length == 1 %}-disabled{% endif %}"{% if search_results|length == 1 and not term %} class="active"{% endif %}><a href="#" id="hier-trigger"{% if not vocab.config.showHierarchy and not search_results|length == 1 %} title="{% trans 'hierarchy-disabled-help' %}"{% endif %}>{% trans "Hier-nav" %}</a></li>
+          {% set disabledHierarchy = not vocab.config.showHierarchy and not search_results|length == 1 %}
+          <h3 class="sr-only">{% trans "Assort vocabulary concepts hierarchically" %}</h3>
+          <li id="hierarchy
+            {%- if disabledHierarchy %}-disabled{% endif -%}
+            "
+            {% if search_results|length == 1 and not term %} class="active"{% endif %}
+          >
+            <a href="#" id="hier-trigger"
+            {% if disabledHierarchy%} title="{% trans 'hierarchy-disabled-help' %}"{% endif %}
+            >{% trans "Hier-nav" %}
+            </a>
+          </li>
           {% if vocab.config.groupClassURI %}
+          <h3 class="sr-only">{% trans "Assort vocabulary concepts and groupings hierarchically" %}</h3>
           <li id="groups"><a href="{{ request.vocabid }}/{{ request.lang }}/groups{% if request.contentLang != request.lang %}?clang={{ request.contentLang }}{% endif %}">{% trans "Group-nav" %}</a></li>
           {% endif %}
           {% if vocab.config.showChangeList %}
+            <h3 class="sr-only">{% trans "Assort vocabulary concepts by most recent changes" %}</h3>
             <li id="changes"><a href="{{ request.vocabid }}/{{ request.lang }}/new{% if request.contentLang != request.lang %}?clang={{ request.contentLang }}{% endif %}">{% trans "Changes-nav" %}</a></li>
           {% endif %}
         {% else %}
+          <h3 class="sr-only">{% trans "Search options tab" %}</h3>
           <li id="limit"><p>{% trans "Search options" %}</p></li>
         {% endif %}
         </ul>
       </div>
       {% if letters %}
+        <h4 class="sr-only">{% trans "Choose alphabetical listing letter" %}</h4>
```

Comment:
As a side note - it is a shame that the screen reader annotations like these only work when traversing the headers on the page with a H-key, but not when traversing clickable items with a Tab-key. However, I really think this is an issue with the screen reader, not with the Skosmos side bar's structure.

Reply:
Yes. There are ways to make this work, e.g., using aria-labelledBy or similar, I think. One solution could be changing the element structure, too. I am not an expert on this subject so I would like to learn more of the best practices how to implement such features. One thing to consider is compatibility - ARIAs may not be fully supported by every reader?